salt/modules/file.py

```diff
@@ -2189,7 +2189,7 @@ def replace(path,
                 raise CommandExecutionError("Exception: {0}".format(exc))
             # write new content in the file while avoiding partial reads
             try:
-                fh_ = salt.utils.atomicfile.atomic_open(path, 'w')
+                fh_ = salt.utils.atomicfile.atomic_open(path, 'wb')
```

Comment:
This is going to break. You're opening the file as binary and then writing strings, not bytestrings, in the loop below. Why did you make this change, maybe I can help if I understand what the reason is.